content/docs/learning-guide/tutorials/extend-the-editor/shape-example-cpp.md

```diff
@@ -0,0 +1,543 @@
+            EditorComponentAPIBus::BroadcastResult(
+                componentNames,
+                &EditorComponentAPIRequests::FindComponentTypeNames,
+                AZ::ComponentTypeList{ typeId }
+            );
+
+            AZ_Assert(componentNames.size() == 1, "Unable to find component name");
+
+            QString shapeName(componentNames[0].c_str());
+            shapeName.replace(" ", "");
+
+            entityName = QString("%1_%2").arg(entityName).arg(shapeName);
+        }
+        
+        EditorEntityAPIBus::Event(newEntityId, &EditorEntityAPIRequests::SetName, entityName.toUtf8().constData());
+    }
+
+    // 3
+    bool validFloat = false;
+    float scale = m_scaleInput->currentText().toFloat(&validFloat);
+    if (validFloat)
+    {
+        AZ::TransformBus::Event(newEntityId, &AZ::TransformInterface::SetLocalUniformScale, scale);
+    }
+
+    // 4
+    EditorComponentAPIBus::Broadcast(&EditorComponentAPIRequests::AddComponentsOfType, newEntityId, AZ::ComponentTypeList{ typeId });
+}
+```
+
+
+## Icon
+
+An icon is an image file that's used to represent your tool in the Editor. The icon appears in the Edit Mode Toolbar in the Editor (see the following image).
+
+{{< image-width "/images/learning-guide/tutorials/extend-the-editor/shape-example/icon.png" "500" "Add an icon for your tool in the Editor" >}}
+
+
+### Add an icon
+
+The following instructions walk you through how to store the icon using the Qt Resource System and load it from your Gem module. Be aware that some of the instructions may already be done by the `PythonToolGem` template.
+
+1. Add an image file to the `Code/Source` directory to use as your icon. In this example, the icon is named `toolbar.svg`. We recommend that your icon adheres to the guidelines in [UI development best practices](https://o3de.org/docs/tools-ui/uidev-component-development-guidelines/#ui-development-best-practices).
+
+2. Add your icon to PyShapeExample Gem's resources by updating `Code/Source/ShapeExample.qrc` with your new icon's file name.
+    
+    ```xml
+    <!DOCTYPE RCC><RCC version="1.0">
+        <qresource prefix="/PyShapeExample">
+            <file alias="toolbar_icon.svg">toolbar_icon.svg</file>
+        </qresource>
+    </RCC>
+    ```
+
+3. Register ShapeExample Gem's resources to Qt Resource System by adding the following code in `Code/Source/EditorModule.cpp`.
+
+    - Define the function `InitShapeExampleResource()` and call `Q_INIT_RESOURCE(...)` to register the Qt resrouces listed in `ShapeExample.qrc`.
+
+    ```cpp
+    void InitShapeExampleResources()
+    {
+        Q_INIT_RESOURCE(ShapeExample);
+    }
+    ```
+
+   - Call `InitShapeExampleResource()` in the `EditorModule` class's constructor.
+
+## Build and debug your tool
+
+Build and debug your custom tool using **Visual Studio 2019**.
+
+1. Launch Visual Studio 2019 and open the O3DE solution.
+
+2. Find your Gem in the Solution Explorer. Right-click and select **Build** or **Debug**.
```

Comment:
I don't think we need steps for Visual Studio, specifically. After all, they could just build again from the command line. They might also be using a different IDE for Linux. Also, there isn't really any information provided here about debugging. Maybe the title could just be named "Build and test your tool". Debugging is sort of implied.

Reply:
I lean towards keeping the steps in the section in because it shows how to build only the Gem, and not the entire engine, which saves time during development. At the moment, I only tested this on Windows and using Visual Studios.

For now, I can specify that these are for building with Windows, and I can change the title to "Build and test your Gem". (Changing "tool" to "Gem".)